components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/primary/PrimaryCardView.kt

```diff
@@ -15,4 +15,64 @@
  */
 package uk.gov.hmrc.components.compose.organism.primary
 
-class PrimaryCardView
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import uk.gov.hmrc.components.compose.atom.heading.Heading5
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+
+@Composable
+fun PrimaryCard(
+    modifier: Modifier = Modifier,
+    title: String,
+    titleContentDescription: String = "",
+    childPadding: Boolean = true,
+    content: @Composable (ColumnScope.() -> Unit)? = null
+) {
+    val childPaddingState by remember { derivedStateOf { childPadding } }
+
+    Column(
+        modifier = modifier
+            .background(HmrcTheme.colors.hmrcWhiteBackground)
+            .fillMaxWidth()
+    ) {
+        Heading5(
+            text = title,
+            modifier = Modifier
+                .padding(horizontal = HmrcTheme.dimensions.hmrcSpacing16)
+                .padding(top = HmrcTheme.dimensions.hmrcSpacing16)
+                .semantics(mergeDescendants = true) {
+                    contentDescription = titleContentDescription
+                }
+                .fillMaxWidth()
+        )
+        if (content != null) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .then(
+                        if (childPaddingState) {
```

Comment:
Should the `childPaddingState` set at a higher level, covering the whole component, rather then only covering the `content` part of the component?

Reply:
I don't think so.. the child padding is meant for automatically adding padding to eg. text or buttons added in the content section of this card, but we need to be able to remove the default padding if we want to add eg. a secondary button that needs to stretch to the edges. 
So I think it's correct for the child padding to only apply to the children (content).